charts/azure-aks-aso/templates/credentials.yaml

```diff
@@ -23,4 +23,7 @@ stringData:
   {{- if .Values.authMode }}
   AUTH_MODE: {{ .Values.authMode | quote }}
   {{- end }}
+  {{- if .Values.useWorkloadIdentity }}
+  USE_WORKLOAD_IDENTITY_AUTH: {{ .Values.useWorkloadIdentity | quote }}
+  {{- end }}
```

Comment:
ASO doesn't know how to make use of this `USE_WORKLOAD_IDENTITY_AUTH` variable in this Secret, so it should probably be removed. That field is only read on the global aso-controller-settings Secret, not these per-resource credential Secrets.